packages/eas-cli/src/commands/metadata.ts

```diff
@@ -0,0 +1,83 @@
+import { getConfig } from '@expo/config';
+import { Errors, Flags } from '@oclif/core';
+
+import EasCommand from '../commandUtils/EasCommand';
+import Log from '../log';
+import { createMetadataContextAsync } from '../metadata/context';
+import { handleMetadataError } from '../metadata/errors';
+import { uploadMetadataAsync } from '../metadata/upload';
+import { RequestedPlatform } from '../platform';
+import { findProjectRootAsync, getProjectIdAsync } from '../project/projectUtils';
+
+type RawCommandFlags = {
+  platform?: string;
+  profile?: string;
+  'non-interactive': boolean;
+};
+
+type CommandFlags = {
+  requestedPlatforms: RequestedPlatform;
+  profile?: string;
+  nonInteractive: boolean;
+};
+
+export default class Metadata extends EasCommand {
+  static description = 'upload metadata configuration to the app stores';
+
+  static flags = {
+    platform: Flags.enum({
+      char: 'p',
+      options: ['ios'],
+    }),
+    profile: Flags.string({
+      description:
+        'Name of the submit profile from eas.json. Defaults to "production" if defined in eas.json.',
+    }),
+    'non-interactive': Flags.boolean({
+      default: false,
+      description: 'Run command in non-interactive mode',
+    }),
+  };
+
+  static args = [];
+
+  async runAsync(): Promise<void> {
+    const { flags: rawFlags } = await this.parse(Metadata);
+    const flags = await this.sanitizeFlagsAsync(rawFlags);
+
+    const projectDir = await findProjectRootAsync();
+    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    await getProjectIdAsync(exp);
+
+    const metadataContext = await createMetadataContextAsync({
+      projectDir,
+      exp,
+      profileName: flags.profile,
+      nonInteractive: flags.nonInteractive,
+    });
+
+    try {
+      Log.addNewLineIfNone();
+      await uploadMetadataAsync(metadataContext);
+      Log.addNewLineIfNone();
```

Comment:
Doesn't need to change but I don't like using these `addNewLineIfNone` since they add some sort of uncertainty about the styling. The use of this method implies there is some global side-effect we are accounting for, this makes it harder to reuse code.

Very minor, but I'd personally opt towards just logging a new line.